File: server/models/document.go
package models

import (
	"time"
)

type Document struct {
	ID         uint64    `gorm:"primaryKey" json:"id"`
	Title      string    `gorm:"size:256;not null" json:"title"`
	Content    string    `gorm:"type:text" json:"content"`
	Summary    string    `gorm:"type:text" json:"summary"`
	CategoryID *uint64   `gorm:"index" json:"category_id"`
	UserID     uint64    `gorm:"index;not null" json:"user_id"`
	Status     string    `gorm:"size:20;default:draft" json:"status"`
	ViewCount  int       `gorm:"default:0" json:"view_count"`
	IsFavorite bool      `gorm:"default:false" json:"is_favorite"`
	Version    int       `gorm:"default:1" json:"version"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`

	// 关联
	Category *Category `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Tags     []*Tag    `gorm:"many2many:document_tags;" json:"tags,omitempty"`
	User     *User     `gorm:"foreignKey:UserID" json:"-"`
}

func (Document) TableName() string {
	return "documents"
}

File: web/src/api/document.ts
import request from "./request";

export interface Document {
  id: number;
  title: string;
  content: string;
  summary: string;
  category_id: number | null;
  status: string;
  view_count: number;
  is_favorite: boolean;
  version: number;
  category?: {
    id: number;
    name: string;
  };
  tags: { id: number; name: string }[];
  created_at: string;
  updated_at: string;
}

export interface PaginatedResult<T> {
  list: T[];
  pagination: {
    page: number;
    page_size: number;
    total: number;
  };
}

export interface CreateDocumentParams {
  title: string;
  content: string;
  summary?: string;
  category_id?: number | null;
  tag_ids?: number[];
  status?: string;
}

export interface UpdateDocumentParams extends CreateDocumentParams {
  is_favorite?: boolean;
}

export interface ListDocumentsParams {
  page?: number;
  page_size?: number;
  category_id?: number;
  status?: string;
}

export function listDocuments(params?: ListDocumentsParams) {
  return request.get<any, PaginatedResult<Document>>("/documents", { params });
}

export function searchDocuments(keyword: string, page = 1, pageSize = 20) {
  return request.get<any, PaginatedResult<Document>>("/documents/search", {
    params: { keyword, page, page_size: pageSize },
  });
}

export function getDocument(id: number) {
  return request.get<any, Document>(`/documents/${id}`);
}

export function createDocument(params: CreateDocumentParams) {
  return request.post<any, Document>("/documents", params);
}

export function updateDocument(id: number, params: UpdateDocumentParams) {
  return request.put<any, Document>(`/documents/${id}`, params);
}

export function deleteDocument(id: number) {
  return request.delete(`/documents/${id}`);
}
